app/main.py

```diff
@@ -1,3 +1,57 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import os
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open(os.path.join("app", "config.json"), "r") as f:
+        all_data = json.load(f)
+
+    customers_data = all_data["customers"]
+    shops_list = [
+        Shop(shop["name"], shop["products"], shop["location"])
+        for shop in all_data["shops"]
+    ]
+    fuel_price = all_data["FUEL_PRICE"]
+    for customer_data in customers_data:
+        current_customer = Customer(
+            customer_data["name"],
+            customer_data["money"],
+            customer_data["product_cart"],
+            customer_data["location"],
+            customer_data["car"],
+        )
+
+        print(f"{current_customer.name} has {current_customer.money} dollars")
+
+        chosen_shop = current_customer.choose_shop(shops_list, fuel_price)
+        if not isinstance(chosen_shop, Shop):
+            print(
+                f"{current_customer.name} doesn't have enough money"
+                f" to make a purchase in any shop"
+            )
+            continue
+        print(f"{current_customer.name} rides to {chosen_shop.name}\n")
+        print("Date: 04/01/2021 12:33:41")
+        print(f"Thanks, {current_customer.name}, for your purchase!")
+        print("You have bought: ")
```

Comment:
Use one print, you can make a new line using `\n` inside a string. Fix everywhere